### PR TITLE
BA-26 search functionality

### PR DESCRIPTION
**Pull request by antkiewiczk**

### CODE REVIEW

File: apps/bajour/pages/search/index.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+              style={{marginTop: '20px'}}
+            />
+          )}
+        </>
+      )}
+    </SearchPageWrapper>
+  )
+}
+
+export const NavbarPhraseResults = ({teasers}: {teasers: any[]}) => {
```
Comment (Itrulia) [inline apps/bajour/pages/search/index.tsx:142]: Why any?

File: apps/bajour/pages/search/index.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+      {error && <SearchStatus>Error: {error.message}</SearchStatus>}
+      {phraseResultTeasers.length > 0 && (
+        <>
+          <NavbarPhraseResults teasers={phraseResultTeasers} />
+          {totalCount > ITEMS_PER_PAGE && (
+            <Pagination
+              count={Math.ceil(totalCount / ITEMS_PER_PAGE)}
+              page={page}
+              onChange={handlePageChange}
+              style={{marginTop: '20px'}}
```
Comment (Itrulia) [inline apps/bajour/pages/search/index.tsx:133]: No inline style please

File: apps/bajour/pages/search/index.tsx (hunk 1 above)
Comment (antkiewiczk) [reply apps/bajour/pages/search/index.tsx:142]: I tried using AricleList[] | PageList[] and making it strongly typed but there were weird typescript issues and after some time I got tired of them and used `any`

Comment (Itrulia) [reply apps/bajour/pages/search/index.tsx:145]:
```diff
diff --git a/apps/bajour/pages/search/index.tsx b/apps/bajour/pages/search/index.tsx
index d0e3599d..f31ccf97 100644
--- a/apps/bajour/pages/search/index.tsx
+++ b/apps/bajour/pages/search/index.tsx
@@ -1,6 +1,6 @@
 import {css, Pagination, styled, TextField} from '@mui/material'
 import {Button} from '@wepublish/ui'
-import {alignmentForTeaserBlock, ApiV1, Teaser} from '@wepublish/website'
+import {alignmentForTeaserBlock, ApiV1, articleToTeaser, Teaser} from '@wepublish/website'
 import {useRouter} from 'next/router'
 import {useEffect, useState} from 'react'
 import {MdSearch} from 'react-icons/md'
@@ -53,6 +53,16 @@ export const SearchPagination = styled(Pagination)`
   margin-top: 20px;
 `
 
+export const pageToTeaser = (page: ApiV1.Page): ApiV1.PageTeaser => ({
+  __typename: 'PageTeaser',
+  style: ApiV1.TeaserStyle.Default,
+  page,
+  image: null,
+  lead: null,
+  preTitle: null,
+  title: null
+})
+
 const ITEMS_PER_PAGE = 5
 
 const SearchPage: React.FC = () => {
@@ -103,15 +113,8 @@ const SearchPage: React.FC = () => {
   const totalPagesCount = phraseData?.phrase?.pages?.totalCount || 0
   const totalCount = totalArticlesCount + totalPagesCount
 
-  const modifiedArticlesNodes = articlesNodes.map(node => ({
-    __typename: 'ArticleTeaser',
-    article: node
-  }))
-
-  const modifiedPagesNodes = pagesNodes.map(node => ({
-    __typename: 'PageTeaser',
-    page: node
-  }))
+  const modifiedArticlesNodes = articlesNodes.map(node => articleToTeaser(node as ApiV1.Article))
+  const modifiedPagesNodes = pagesNodes.map(node => pageToTeaser(node as ApiV1.Page))
 
   const phraseResultTeasers = [...modifiedArticlesNodes, ...modifiedPagesNodes]
 
@@ -142,7 +145,7 @@ const SearchPage: React.FC = () => {
   )
 }
 
-export const NavbarPhraseResults = ({teasers}: {teasers: any[]}) => {
+export const NavbarPhraseResults = ({teasers}: {teasers: ApiV1.Teaser[]}) => {
   return (
     <NavbarPhraseResultsWrapper>
       {teasers.map((teaser, index) => {

```

This turns it into an actual teaser.

File: apps/bajour/pages/search/index.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+`
+
+export const SearchStatus = styled('div')`
+  margin: 0 auto;
+  padding: ${({theme}) => theme.spacing(4)};
+  text-align: center;
+`
+
+export const SearchPagination = styled(Pagination)`
+  margin-top: 20px;
```
Comment (Itrulia) [inline apps/bajour/pages/search/index.tsx:53]: can we use spacing(2) or spacing(3) here?

File: apps/bajour/pages/search/index.tsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    }
+  })
+
+  useEffect(() => {
+    if (query.q) {
+      setPhraseQuery(query.q as string)
+    }
+  }, [query.q])
+
+  const handleSearch = (e: React.FormEvent) => {
```
Comment (Itrulia) [inline apps/bajour/pages/search/index.tsx:83]: Please import the types and not rely on a global hidden import (React) that comes from a babel plugin/compiler.

File: apps/bajour/pages/search/index.tsx [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+  margin-top: 20px;
+`
+
+const ITEMS_PER_PAGE = 5
+
+const SearchPage: React.FC = () => {
+  const router = useRouter()
+  const {query} = router
+  const [rawQuery, setRawQuery] = useState((query.q as string) || '')
+  const [phraseQuery, setPhraseQuery] = useState((query.q as string) || '')
```
Comment (Itrulia) [inline apps/bajour/pages/search/index.tsx:62]: This state is now handled twice, once in the URL and once in a useState. We can just use a single source of truth for the state (URL).

File: libs/website/api/src/lib/schemas/phrase.graphql [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        hasNextPage
+        hasPreviousPage
+      }
+
+      totalCount
+    }
+
+    pages {
+      nodes {
+        ...FullPage
```
Comment (Itrulia) [inline libs/website/api/src/lib/schemas/phrase.graphql:20]: Probably want PageWithoutBlocks

**Review by Itrulia — APPROVED** [review]

**Review by Itrulia — APPROVED** [review]

**Pull request merged**